src/components/TaskCard/TaskCard.jsx

```diff
@@ -1,30 +1,54 @@
+import classNames from "classnames";
 import PropTypes from "prop-types";
 import { useDispatch } from "react-redux";
-import { deleteTask } from "@features/task/taskSlice";
-import formatDate from "@helpers/formatting/formatDate";
 import IconButton from "@components/ui/form/IconButton";
-import { TrashIcon } from "@components/ui/icons";
+import { CheckIcon, TrashIcon } from "@components/ui/icons";
+import { deleteTask, markAsDone } from "@features/task/taskSlice";
+import formatDate from "@helpers/formatting/formatDate";
+import { compareDates } from "@helpers/operations/compareDates";
 import "@components/TaskCard/TaskCard.scss";
 
 function TaskCard({ task }) {
-    const { id, title, createdAt } = task;
+    const { id, title, createdAt, isCompleted, completedAt } = task;
     const dispatch = useDispatch();
 
+    const currentDate = new Date().toISOString();
+    const daysElapsed = compareDates(currentDate, completedAt);
+    const pluralizedDayText = daysElapsed === 1 ? "day" : "days";
+
     const handleDelete = () => {
         dispatch(deleteTask(id));
     };
 
+    const handleMarkAsDone = () => {
+        dispatch(markAsDone(id));
+    };
+
     return (
-        <div className="task-card">
+        <div
+            className={classNames("task-card", {
+                "task-card--complete": isCompleted,
+            })}
+        >
             <h2 className="task-card__title">{title}</h2>
             <span>Created At: {formatDate(createdAt)}</span>
             <div className="task-card__body">
                 <div className="task-card__actions-wrapper">
+                    {!isCompleted && (
+                        <IconButton onClick={handleMarkAsDone}>
+                            <CheckIcon />
```

Comment:
Instead of passing data as children, let's pass the icon as prop into `iconButton`